File: src/Productos/limpieza.java
package Productos;
import java.util.ArrayList;
import java.util.Scanner;
public class limpieza extends Productos{
    ArrayList <limpieza> ProductosLimpieza= new ArrayList<>();
    //si es false no es ecoFriendly si es true si lo es;
    private boolean ecoFriendly;
    Scanner leer=new Scanner(System.in);

    public limpieza(String nombre, double precio, String fechaDeImportacion, int stock, boolean ecoFriendly) {
        super(nombre, precio, fechaDeImportacion, stock);
        this.ecoFriendly=ecoFriendly;
    }
    public void RegistrarProductoLimpieza(String nombre,double precio, String fechaDeImportacion, int stock){
        Boolean band=true;
        System.out.println("el producto es EcoFriendly?, ingrese si o no");
        while(band==true){
        String s=leer.nextLine().toUpperCase();
            if (s.equals("SI")){
                ecoFriendly=true;
                band=false;
            } else if (s.equals("NO")) {
                ecoFriendly=false;
                band=false;
            }else{
                System.out.println("ingrese si o no");
            }
        }
        limpieza productoNuevoLimpieza=new limpieza(nombre, precio, fechaDeImportacion, stock, ecoFriendly);
        ProductosLimpieza.add(productoNuevoLimpieza);
    }
}

File: limpieza.py
import Productos.Productos
class Limpieza(Productos):
    def __init__(self, nombre, precio, fecha_de_importacion, stock, eco_friendly):
        super().__init__(nombre, precio, fecha_de_importacion, stock)
        self.eco_friendly = eco_friendly
        self.productos_limpieza = []
        self.leer = input  # Cambio Scanner por input en Python

    def registrar_producto_limpieza(self, nombre, precio, fecha_de_importacion, stock):
        band = True
        print("¿El producto es EcoFriendly? Ingrese 'si' o 'no'")
        while band:
            s = input().upper()  # Cambio nextLine() por input() en Python
            if s == "SI":
                self.eco_friendly = True
                band = False
            elif s == "NO":
                self.eco_friendly = False
                band = False
            else:
                print("Ingrese 'si' o 'no'")
        nuevo_producto_limpieza = Limpieza(nombre, precio, fecha_de_importacion, stock, self.eco_friendly)
        self.productos_limpieza.append(nuevo_producto_limpieza)